samples-final/2.AspNetCore-ConversationUpdate-Bot/ConversationUpdateBot.cs

```diff
@@ -17,17 +17,28 @@ public Task OnTurn(ITurnContext context)
             {
                 // On "conversationUpdate"-type activities this bot will send a greeting message to users joining the conversation.
                 case ActivityTypes.ConversationUpdate:
-                    var newUserName = context.Activity.MembersAdded.FirstOrDefault()?.Name;
-                    if (!string.IsNullOrWhiteSpace(newUserName) && newUserName != "Bot")
+
+                    // Both the bot and the user will be in Activity.MembersAdded.  We only want to send a message to the user.
+                    // The user is not the recipient, the bot is in this case. This is what we are checking for here.
+                    if (context.Activity.MembersAdded.Any(member => member.Id != context.Activity.Recipient.Id))
                     {
-                        return context.SendActivity($"Hello {newUserName}!");
+                        var newUserName = context.Activity.MembersAdded.FirstOrDefault()?.Name;
```

Comment:
This is still only working with the first user in the list. For a group conversation, there could be multiple users added at the same time. You could send one group message welcoming all the new users on the conversation, or you could send individual direct messages to each new user. I'm not sure this sample shows the best practice as is.